Logger/BaseLogger.cs

```diff
@@ -2,6 +2,13 @@
 {
     public abstract class BaseLogger
     {
+        public string ClassName { get; set; } = string.Empty;
```

Comment:
```suggestion
        public string? ClassName { get; set; }
```
No need to set this to an empty string. You can leave it as nullable.